tests/runtime/client/test_remote_model_initializer.py

```diff
@@ -521,3 +523,90 @@ def test_remote_initializer_retry(sample_task_model_id, open_port, protocol):
                 request_id=random_test_id(),
                 throw_first_num_requests=5,
             )
+
+
+@pytest.mark.parametrize("protocol", ["grpc", "http"])
+def test_remote_initializer_always_new_channel(
+    sample_task_model_id, open_port, protocol
+):
+    """Test to ensure RemoteModule Initializer negative max connection age always creates
+    a new channel"""
+    local_module_class = (
+        ModelManager.get_instance().retrieve_model(sample_task_model_id).__class__
+    )
+
+    # Construct Remote Module Config with a max session age of 0
+    connection_info = ConnectionInfo(
+        hostname="localhost", port=open_port, max_session_age=0
+    )
+    remote_config = RemoteModuleConfig.load_from_module(
+        local_module_class,
+        connection_info,
+        protocol,
+        MODEL_MESH_MODEL_ID_KEY,
+        sample_task_model_id,
+    )
+    # Set random module_id so tests don't conflict
+    remote_config.module_id = random_test_id()
+
+    with runtime_test_server(open_port, protocol=protocol):
+        # Construct initializer and RemoteModule
+        remote_initializer = RemoteModelInitializer(Config({}), "test")
+        remote_model = remote_initializer.init(remote_config)
+        assert isinstance(remote_model, ModuleBase)
+
+        # Run RemoteModule Request and ensure that the remote model does not have a saved channel
+        model_result = remote_model.run(
+            SampleInputType(name="Test"),
+            request_id=random_test_id(),
+        )
+        assert model_result.greeting == "Hello Test"
+        assert not remote_model._conn_channel
+
+
+@pytest.mark.parametrize("protocol", ["grpc", "http"])
+def test_remote_initializer_always_time_new_channel(
+    sample_task_model_id, open_port, protocol
+):
+    """Test to ensure RemoteModule Initializer max session age correctly takes affect"""
+    local_module_class = (
+        ModelManager.get_instance().retrieve_model(sample_task_model_id).__class__
+    )
+
+    # Construct Remote Module Config with a small max session age
+    connection_info = ConnectionInfo(
+        hostname="localhost", port=open_port, max_session_age=0.5
+    )
+    remote_config = RemoteModuleConfig.load_from_module(
+        local_module_class,
+        connection_info,
+        protocol,
+        MODEL_MESH_MODEL_ID_KEY,
+        sample_task_model_id,
+    )
+    # Set random module_id so tests don't conflict
+    remote_config.module_id = random_test_id()
+
+    with runtime_test_server(open_port, protocol=protocol):
+        # Construct initializer and RemoteModule
+        remote_initializer = RemoteModelInitializer(Config({}), "test")
+        remote_model = remote_initializer.init(remote_config)
+        assert isinstance(remote_model, ModuleBase)
+
+        # Run RemoteModule Request twice to ensure the channel is retained. Run the second one with a small sleep to ensure the third request gets a new object
```

Comment:
We need a new definition of "small sleep!" I worry this is a little error prone with sleep-based behavior. Could we use a mock to patch over the `time.time` function to make it appear as "timed out" on the third call?

Reply:
Good call on mock! Done